implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -83,14 +95,14 @@ static String prepareEndpoint(String uri) {
                 endpoint = uri;
             } else {
                 try {
-                    if (uri.contains("localhost") || uri.contains("127.0.0.1")) {
+                    if (uri.contains("localhost") || uri.contains("127.0.0.1") || uri.contains("::1")) {
```

Comment:
You could also parse it into a URL and use `getHost` (for the unlikely event that the path contains localhost or something like that).